rest/alert_test.go

```diff
@@ -0,0 +1,562 @@
+			linkHeader.Set("Link", `</alerting/v1/alerts?next=`+*alertList[1].Name+`>; rel="next"`)
+			require.Nil(t, mock.AddAlertListTestCase("", nil, linkHeader, alertList[0:2]))
+			require.Nil(t, mock.AddAlertListTestCase("next="+*alertList[1].Name, nil, nil, alertList[2:5]))
+
+			respAlerts, _, err := client.Alerts.List()
+			require.Nil(t, err)
+			compareAlertLists(t, alertList, respAlerts)
+		})
+
+		t.Run("List without pagination", func(t *testing.T) {
+			defer mock.ClearTestCases()
+
+			require.Nil(t, mock.AddAlertListTestCase("", nil, nil, alertList))
+
+			respAlerts, _, err := client.Alerts.List()
+			require.Nil(t, err)
+			require.NotNil(t, respAlerts)
+
+			compareAlertLists(t, alertList, respAlerts)
+		})
+	})
+
+	t.Run("Get", func(t *testing.T) {
+
+		t.Run("Success", func(t *testing.T) {
+			defer mock.ClearTestCases()
+
+			client.FollowPagination = true
+			alert := alertList[0]
+			require.Nil(t, mock.AddAlertGetTestCase(*alert.ID, nil, nil, alert))
+
+			respAlert, _, err := client.Alerts.Get(*alert.ID)
+			require.Nil(t, err)
+			require.NotNil(t, respAlert)
+			compareAlerts(t, alert, respAlert)
+		})
+		t.Run("Not Found", func(t *testing.T) {
+			defer mock.ClearTestCases()
+
+			require.Nil(t, mock.AddAlertFailTestCase(
+				http.MethodGet, "abcd-efgh-ijkl", http.StatusNotFound,
+				nil, nil, `{"message": "test error"}`,
+			))
+
+			respAlert, resp, err := client.Alerts.Get("abcd-efgh-ijkl")
+			require.Nil(t, respAlert)
+			require.NotNil(t, err)
+			require.Contains(t, err.Error(), "test error")
+			require.Equal(t, http.StatusNotFound, resp.StatusCode)
+		})
+
+		t.Run("Other Error", func(t *testing.T) {
+			c := api.NewClient(errorClient{}, api.SetEndpoint(""))
+			respAlert, resp, err := c.Alerts.Get("abcd-efgh-ijkl")
+			require.Nil(t, resp)
+			require.Error(t, err)
+			require.Nil(t, respAlert)
+		})
+	})
+
+	t.Run("Create", func(t *testing.T) {
+
+		t.Run("Success", func(t *testing.T) {
+			defer mock.ClearTestCases()
+
+			alertToCreate := alerting.Alert{
+				ID:   strPtr("f136f0cd-eca8-4b2b-9b2d-9a6631200d51"),
+				Data: json.RawMessage(nil),
+				Name: strPtr("first_alert"),
+				NotifierListIds: []string{
+					"66d07ca6e113eb00014fe257",
+					"66d07caf8519c000011cdda6",
+					"6707da567cd4f300012cd7e4",
+				},
+				RecordIds: []string{},
+				Type:      strPtr("zone"),
+				Subtype:   strPtr("transfer_failed"),
+				ZoneNames: []string{
+					"alerttest.com", "alerttest.net",
+				},
+			}
+
+			alertResponse := alertList[0]
+
+			// Pass by value here. alertToCreate will be modified by client.Alerts.Create
+			require.Nil(t, mock.AddAlertCreateTestCase(nil, nil, alertToCreate, *alertResponse))
+
+			_, err := client.Alerts.Create(&alertToCreate)
+			require.Nil(t, err)
+			// alerttoCreate will be modified by the call to Create and fully populated.
+			compareAlerts(t, alertResponse, &alertToCreate)
+		})
+
+		// TODO Add test for error response when updating fields that can't be changed.
```

Comment:
Is this difficult? Why isn't it in the PR?

Reply:
Removed, it's not difficult it's also not an issue on alert creation - I copied this from either update or replace which does test this.